media/admin/com_virtuemart/models/zasilkovna_src/VirtueMartModelZasilkovna/ConfigurationValidator.php

```diff
@@ -0,0 +1,156 @@
+<?php
+namespace VirtueMartModelZasilkovna;
+
+class ConfigurationValidator
+{
+    public const KEY_USE_DEFAULT_WEIGHT = 'zasilkovna_use_default_weight';
+    public const KEY_DEFAULT_WEIGHT = 'zasilkovna_default_weight';
+    public const KEY_USE_DEFAULT_DIMENSIONS = 'zasilkovna_use_default_dimensions';
+    public const KEY_API_PASS = 'zasilkovna_api_pass';
+    public const KEY_ESHOP_LABEL = 'zasilkovna_eshop_label';
+    public const KEY_DEFAULT_LENGTH = 'zasilkovna_default_length';
+    public const KEY_DEFAULT_WIDTH = 'zasilkovna_default_width';
+    public const KEY_DEFAULT_HEIGHT = 'zasilkovna_default_height';
+    public const KEY_PAYMENT_METHOD_PREFIX = 'zasilkovna_payment_method_';
+    public const CONFIG_DEFAULTS = [
+        self::KEY_USE_DEFAULT_WEIGHT => false,
+        self::KEY_DEFAULT_WEIGHT => '',
+        self::KEY_USE_DEFAULT_DIMENSIONS => false,
+        self::KEY_DEFAULT_LENGTH => '',
+        self::KEY_DEFAULT_WIDTH => '',
+        self::KEY_DEFAULT_HEIGHT => '',
+        self::KEY_API_PASS => '',
+        self::KEY_ESHOP_LABEL => '',
+    ];
+
+
+    /**
+     * @var array<string, mixed>
+     */
+    private array $configuration;
+
+    /**
+     * @var array <string, string>|array<string, array<string>>
+     */
+    private array $errors = [];
+    
+    /**
+     * @var array<string, mixed>
+     */
+    private array $validData = [];
+
+    /**
+     * @param array<string, mixed> $configuration
+     */
+    public function __construct(array $configuration)
+    {
+        $this->configuration = $configuration;
+    }
+
+    public function validate(): void {
+        $this->validateApiPassword();
+        $this->validateWeight();
+        $this->validateDimensions();
+        $this->setValidDataForNotValidatedFields();
```

Comment:
Normalizace dat by neměla probíhat současně s validací, je to míchání odpovědností.
Vytvoř místo toho metodu normalize(), která postupně sestaví $normalizedData a rovnou je vrátí, místo   $this->validData a getValidData(), které se stejně volají jen jednou.